core/common/src/main/java/alluxio/metrics/MultiValueMetricsAggregator.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.metrics;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An interface for the aggregator that aggregates metrics of different hosts into multiple
+ * cluster-level metrics.
+ */
+public interface MultiValueMetricsAggregator {
```

Comment:
Can we combine this interface with `MetricsAggregator`?

Reply:
We could. `MetricsAggregator ` is a special case of `MetricsAggregator` and therefore in the implementation it has some optimizations. For example, it does not call the `registerGaugeIfAbsent` per heartbeat, but calls it once upon start.

Reply:
Do you mean `MultiValueMetricsAggregator` is a special case of `MetricsAggregator`? We don't need to combine the interfaces if you think it makes more sense to establish a relationship between them. Currently there is no relation so it is awkward.

Reply:
I think MetricsAggregator can be categorized as `MultiValueMetricsAggregator ` and `SingleValueAggregator`. However, they do not share some common methods.